kotlinx-coroutines-core/common/src/flow/Flow.kt

```diff
@@ -108,29 +126,57 @@ import kotlin.coroutines.*
  *  - Collecting another flow from a separate context is allowed, but it has the same effect as
  *    [flowOn] operator on that flow, which is more efficient.
  *
+ * ### Exception transparency
+ *
+ * Flow implementations never catch or handle exceptions that occur in downstream flows. From implementation standpoint
+ * it means that calls to [emit][FlowCollector.emit] and [emitAll] shall never be wrapped into
+ * `try { ... } catch { ... }` blocks. Exception handling in flows shall be performed with
+ * [catch][Flow.catch] operator and it is designed to catch only exception coming from upstream flow, while passing
+ * all the downstream exceptions. Similarly, terminal operators like [collect][Flow.collect]
+ * throw any unhandled exception that occurs in its code or in upstream flows, for example:
+ *
+ * ```
+ * flow { emitData() }
+ *     .map { computeOne(it) }
+ *     .catch { ... } // catches exceptions in emitData and computeOne
+ *     .map { computeTwo(it) }
+ *     .collect { process(it) } // throws exceptions from process and computeTwo
+ * ```
+ *
+ * Failure to adhere to the exception transparency requirement would strange behaviours that would make it hard
```

Comment:
>  Failure to adhere to the exception transparency requirement **_would strange behaviours_** that would make it hard

Is this correct?

Otherwise a good improvement to the documentation.

Reply:
Thanks for noticing. That was a typo (missing words, actually). I've also improved this whole sentence a bit having reread it from scratch.